src/Exception/BatchException.php

```diff
@@ -2,61 +2,136 @@
 
 namespace Http\Client\Exception;
 
+use Http\Client\BatchResult;
+use Http\Client\Exception;
+use Psr\Http\Message\RequestInterface;
+
 /**
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
  */
-final class BatchException extends TransferException
+final class BatchException extends RuntimeException
 {
     /**
-     * @var TransferException[]
+     * @var BatchResult
+     */
+    private $result;
+
+    /**
+     * @var \SplObjectStorage
      */
     private $exceptions;
 
+    public function __construct()
+    {
+        $this->exceptions = new \SplObjectStorage();
+    }
+
     /**
-     * @param TransferException[] $exceptions
+     * Returns a list BatchResult which contains succesful responses
+     *
+     * @return BatchResult
      */
-    public function __construct(array $exceptions = [])
+    public function getResult()
     {
-        parent::__construct('An error occurred when sending multiple requests.');
+        return $this->result;
+    }
 
-        foreach ($exceptions as $e) {
-            if (!$e instanceof TransferException) {
-                throw new InvalidArgumentException('Exception is not an instanceof Http\Client\Exception\TransferException');
-            }
-        }
+    /**
+     * Sets the successful response list
+     *
+     * @param BatchResult $result
+     *
+     * @internal
+     */
+    public function setResult(BatchResult $result)
+    {
+        $this->result = $result;
+    }
+
+    /**
+     * Checks if a request is successful
+     *
+     * @param RequestInterface $request
+     *
+     * @return boolean
+     */
+    public function isSuccessful(RequestInterface $request)
+    {
+        return isset($this->result) && $this->result->hasResponseFor($request);
```

Comment:
It is just a foolproofness check. The best solution would be to inject the BatchResult in the constructor, but it is not possible if the order or processing is linear, since the BatchResult is immutable, so after injecting we cannot modify the internal object. One workaround for this could be to create and empty BatchResult in the exception constructor.